app/backend/approaches/readretrieveread.py

```diff
@@ -90,7 +91,12 @@ def run(self, q: str, overrides: dict[str, Any]) -> Any:
             prefix=overrides.get("prompt_template_prefix") or self.template_prefix,
             suffix=overrides.get("prompt_template_suffix") or self.template_suffix,
             input_variables = ["input", "agent_scratchpad"])
-        llm = AzureOpenAI(deployment_name=self.openai_deployment, temperature=overrides.get("temperature") or 0.3, openai_api_key=openai.api_key)
+        
+        if(self.openai_type == "azure"):
+            llm = AzureOpenAI(deployment_name=self.openai_deployment, temperature=overrides.get("temperature") or 0.3, openai_api_key=openai.api_key)
```

Comment:
`.get()` should take a the default argument as the second parameter, so you can replace `.get("temperature") or 0.3` with `get("temperature", 0.3)` in all occurrences ( I see a few in this PR )